File: code/reviewServer/src/main/java/com/jay/reviewserver/domain/entity/ExternalImage.java
package com.jay.reviewserver.domain.entity;

import jakarta.persistence.Access;
import jakarta.persistence.AccessType;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

@Entity
@DiscriminatorValue("EI")
@Access(AccessType.FIELD)
public class ExternalImage extends ReviewImage {

    @Override
    public String url() {
        return getPath();
    }

    @Override
    public Boolean hasThumbnail() {
        return false;
    }

    @Override
    public String getThumbnail() {
        return "";
    }
}

File: code/productserver/src/main/kotlin/com/jay/productserver/domain/entity/ExternalImage.kt
package com.jay.productserver.domain.entity

import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity

@Entity
@DiscriminatorValue("EI")
class ExternalImage(
    path : String
) : ProductImage(path = path) {
    override fun url(): String {
        return getPath()
    }

    override fun hasThumbnail(): Boolean {
        return false
    }

    override fun thumbnailURL(): String {
        return ""
    }
}